EXPLAINER.md

```diff
@@ -1,118 +1,108 @@
+
+We also then a *need a way for each participant to emit what it did*. We should be able to uniquely identify (within the scope of a single logical operation) the above unit of work done by each participant.
 
-### Trace Flags
-Trace flags communicate information about the trace to remote tracing systems. Currently, the only information transmitted with the trace is whether or not a particular request is "sampled," or captured by a tracing system.
+But the above is not enough: we also need to establish the causality of the work done by different participants. For this, we need a way to capture the parent/child relationships: when a participant calls another, the callee/child needs to know the unique identifier of the caller/parent’s unit of work, so that the child can emit it. How does the child get this? It must be propagated.
 
-## Headers
+If we have all the above, we will gain the ability to connect the dots together.
 
-### Traceparent
+The set of related events emitted by various participants, triggered by the same logical operation, is called a Distributed Trace. The context that is propagated between the participants is referred to as the “trace context” that includes the identifiers described above.
 
-The traceparent header is a request header which contains a trace id, a parent span id, and trace flags. A tracing system transmits the trace ID, span ID, and the ID of each span's parent span, if it exists, with each span to a tracing backend. The tracing backend uses this information to construct the directed acyclic graph which represents the trace. In a system with a single tracing platform, it is the only header necessary to complete a distributed trace.
+### Why does it have to be a standard?
+If there is no standardized way to propagate the trace context, each tracing system would have to create its own method. Typically, this would be in the form of a custom HTTP header. If an operation is handled by multiple participants using different tracing systems, such headers will not be propagated between them and the trace will be broken. When this happens, we will not able to reason about the whole system.
 
-### Tracestate
+To make things interoperable, we need a standard mechanism that everybody can agree to use.
 
-The tracestate header is a request header which allows a tracing platform to transmit platform-specific information, even through systems which are traced using other tracing platforms. Each tracing platform uses a single key, and the value is treated as an opaque token by other tracing platforms. Each tracing platform is required to forward tracestate keys and values.
+### Goals
 
-### Trace Response header
+- To provide a standard mechanism to propagate shared context across various applications and services so that multiple tracing platforms can be used in concert with each other.
+- To provide a method for application developers to trace requests without invading users' privacy.
```

Comment:
Is that an actual goal? We spend a lot of time in the most recent privacy review to explain to reviewers that most privacy concerns are orthogonal to what we are doing in the spec.

Reply:
Since the specification provides guidance on how to generate TraceID and SpanID using mechanisms (generating them using in a pseudo-random way) that respect user's privacy, I think there may still be value in keeping this goal. Please let me know if you disagree.